src/components/tx/SignOrExecuteForm/PermissionsCheck.tsx

```diff
@@ -0,0 +1,290 @@
+  type ChainId,
+  chains,
+  fetchRolesMod,
+  Clearance,
+  type RoleSummary,
+  ExecutionOptions,
+  Status,
+} from 'zodiac-roles-deployments'
+import { OperationType, type MetaTransactionData } from '@safe-global/safe-core-sdk-types'
+import { decodeBytes32String, type JsonRpcProvider } from 'ethers'
+import { KnownContracts, getModuleInstance } from '@gnosis.pm/zodiac'
+
+import { SafeTxContext } from '@/components/tx-flow/SafeTxProvider'
+import { Box, Button, CardActions, Chip, CircularProgress, Divider, Typography } from '@mui/material'
+import commonCss from '@/components/tx-flow/common/styles.module.css'
+import CheckWallet from '@/components/common/CheckWallet'
+import TxCard from '@/components/tx-flow/common/TxCard'
+import { getTransactionTrackingType } from '@/services/analytics/tx-tracking'
+import { TX_EVENTS } from '@/services/analytics/events/transactions'
+import { trackEvent } from '@/services/analytics'
+import useChainId from '@/hooks/useChainId'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import useAsync from '@/hooks/useAsync'
+import WalletRejectionError from './WalletRejectionError'
+import ErrorMessage from '../ErrorMessage'
+import useWallet from '@/hooks/wallets/useWallet'
+import { useWeb3ReadOnly } from '@/hooks/wallets/web3'
+
+const Role = ({ children }: { children: string }) => {
+  let humanReadableRoleKey = children
+  try {
+    humanReadableRoleKey = decodeBytes32String(children)
+  } catch (e) {}
+
+  return <Chip label={humanReadableRoleKey} />
+}
+
+const PermissionsCheck: React.FC<{}> = ({}) => {
+  const chainId = useChainId()
+  const { safeTx, safeTxError } = useContext(SafeTxContext)
+  const [isRejectedByUser, setIsRejectedByUser] = useState<boolean>(false)
+
+  const roles = useRoles(safeTx?.data)
+  const allowingRole = roles.find((role) => role.status === Status.Ok)
+
+  // If a user has multiple roles, we should prioritize the one that allows the transaction's to address (and function selector)
+  const mostLikelyRole =
+    allowingRole ||
+    roles.find((role) => role.status !== Status.TargetAddressNotAllowed && role.status !== Status.FunctionNotAllowed) ||
+    roles.find((role) => role.status !== Status.TargetAddressNotAllowed) ||
+    roles[0]
+
+  const isPending = false
+
+  const handleExecute = async () => {
+    setIsRejectedByUser(false)
+
+    const txId = 'TODO'
+
+    // Track tx event
+    const txType = await getTransactionTrackingType(chainId, txId)
+    trackEvent({ ...TX_EVENTS.EXECUTE_THROUGH_ROLE, label: txType })
+  }
+
+  // Only render the card if the connected wallet is a member of any role
+  if (roles.length === 0) {
+    return null
+  }
+
+  return (
+    <TxCard>
+      <Typography variant="h5">Execute through role</Typography>
+
+      {allowingRole && (
+        <Typography>
+          As a member of the <Role>{allowingRole.roleKey}</Role> role you can execute this transaction immediately.
+        </Typography>
+      )}
+
+      {!allowingRole && (
```

Comment:
What is the use-case behind displaying the role if the user has a not allowing role?

I'd prefer adding a new section "Roles" in the settings where we list all roles with their permissions and only showing the Card in the modal if the user is allowed to use it.

Reply:
Appreciate the feedback. Our thinking was: If a roles member is connected to the Safe web app it's pretty likely they are trying to execute something through the role–especially if the connected wallet is not even an owner on the Safe. For connected wallets that are both, a Safe owner as well as a role member, we could indeed consider hiding the "Execute through role" card if it doesn't pass role permissions checks, even though I'd prefer to still show it for improved transparency.

It's pretty common that permissions become outdated/have not been designed with all cases in mind. In that case it's helpful if the user can see the permission check status to request an update to the role permissions.

____
> I'd prefer adding a new section "Roles" in the settings where we list all roles with their permissions and only showing the Card in the modal if the user is allowed to use it.

We definitely also plan to add a "Roles" section in settings as a next iteration. We gonna share some mock-ups for that soon to kick off the discussion.

Reply:
@jfschwarz 
Thanks for the detailed explanation on the motivation of this. That makes a lot of sense!